include/uprotocol-cpp/uri/builder/BuildEntity.h

```diff
@@ -0,0 +1,101 @@
+/*
+ 
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ 
+ 
+ */
+
+#ifndef UPROTOCOL_CPP_BUILDENTITY_H
+#define UPROTOCOL_CPP_BUILDENTITY_H
+
+#include <algorithm>
+#include <cctype>
+#include <optional>
+#include <string>
+#include <array>
+#include <string_view>
+#include <arpa/inet.h>
+#include <spdlog/spdlog.h>
+#include "../tools/Utils.h"
+#include <uprotocol/uri.pb.h>
+
+namespace uprotocol::uri {
```

Comment:
You need to document your interface and all the methods that you want developers to use, otherwise they will not know what this is for. IDEs pull in the documentation so when developers start typing a method name, they can hover over a method and get the definitions for what the methods do, very important.